### DISCUSSION TITLE
Issue with clickhouse URL deployed on AWS ECS

### DISCUSSION THREAD

**Discussion by Beshr458** [plausible/analytics#2781] [Self-Hosted Support] [answered]
Hi All,

I am deploying plausible on AWS as self hosted type with ECS Fargate. I could get all the containers running but I am facing issues possibly with clickhouse URL. It seems it not supporting DNS name or IPs?

So I have tested setting the clickhouse URL to DNS and public IP and have the security group allowing all traffic and tried in private and public VPCs. But it seems failing to connect.

So the URL looks like this:
For IP: 
`CLICKHOUSE_DATABASE_URL=http://54.160.49.189:8123/plausible_events_db`
For DNS within AWS:
`CLICKHOUSE_DATABASE_URL=http://event-db.plausible.local:8123/plausible_events_db`

And I get the following errors:
````
Loading plausible..
Starting dependencies..
Starting repos..
create Plausible.Repo database if it doesn't exist
create Plausible.ClickhouseRepo database if it doesn't exist
17:49:43.280 [error] Task #PID<0.198.0> started from #PID<0.94.0> terminating
** (DBConnection.ConnectionError) connection not available and request was dropped from queue after 5991ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:

  1. Ensuring your database is available and that you can connect to it
  2. Tracking down slow queries and making sure they are running fast enough
  3. Increasing the pool_size (although this increases resource consumption)
  4. Allowing requests to wait longer by increasing :queue_target and :queue_interval

See DBConnection.start_link/2 for more information

    lib/clickhouse_ecto/connection.ex:79: ClickhouseEcto.Connection.execute/4
    lib/clickhouse_ecto/storage.ex:71: anonymous fn/2 in ClickhouseEcto.Storage.run_query/2
    (elixir 1.13.4) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (elixir 1.13.4) lib/task/supervised.ex:34: Task.Supervised.reply/4
    (stdlib 3.17.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Function: #Function<0.128878517/0 in ClickhouseEcto.Storage.run_query/2>
````

I can create the database manually and locally on the container itself using this command:
`clickhouse-client --query "CREATE DATABASE IF NOT EXISTS plausible_events_db"`

And I can verify it is creating the database correctly by using SHOW DATABASES.

But that will create an empty database with no tables and then the I get 500 error code when registering in plausible UI and see errors in plausible container that is not able to find <database-name>.events table.

So I think the issue all will be solved if I know the syntax of the clickhouse URL.

**Comment by ruslandoga** [comment, marked as answer]
👋 @Beshr458 

Can you try sending a request to either URL with `curl` from the same host that Plausible runs on?

```sh
curl http://54.160.49.189:8123/ping
curl http://event-db.plausible.local:8123/ping
```

**Reply by ruslandoga** [reply]
This would make sure that there is nothing blocking HTTP requests on port 8123 (since `clickhouse-client` uses plain TCP on port 9000 so it's not a completely equivalent test)

**Reply by Beshr458** [reply]
There was an issue with VPC in AWS and once that is fixed, the clickhouse db was reachable.
